packages/protocol/test/unit/utils/StreamPartitionID.test.ts

```diff
@@ -0,0 +1,188 @@
+import SPID  from '../../../src/utils/StreamPartitionID'
+
+describe('StreamPartitionID', () => {
+    const STREAM_ID = 'test-stream-id'
+    const PARTITION = 1
+
+    it('can use object destructuring', () => {
+        const spid = new SPID(STREAM_ID, PARTITION)
+        const { id, partition, key } = spid
+        expect(id).toBe(STREAM_ID)
+        expect(partition).toBe(PARTITION)
+        expect(key).toBe(spid.toString())
```

Comment:
I think it would be useful to actually expect for a literal value of `key` here? In case of a situation where `toString` and `key` would be `undefined` etc. for whatever weird reason.

Reply:
done